numpy/core/src/multiarray/arrayobject.c

```diff
@@ -661,36 +534,42 @@ static PyObject *
 array_repr(PyArrayObject *self)
 {
     PyObject *s, *arglist;
+    PyObject *reprfunc = NULL;
 
-    if (PyArray_ReprFunction == NULL) {
-        s = array_repr_builtin(self, 1);
+    npy_cache_import("numpy.core._internal", "_pyarray_repr_function", &reprfunc);
```

Comment:
Ah ok, it all makes sense now, the None case is wrapped by another layer of python which replaces None, by the default printing function, so that NULL is basically not accessible any more.

I like deletion of that code, it seems really never used, but it is accessible from the C-Api (by setting things to NULL). The only thing I am still a bit wondering about is, whether you can get the NULL case in a C-Program because `numpy/core/numeric.py` might not be imported automatically.